main.tf

```diff
@@ -25,14 +25,14 @@ resource "aws_route" "public_internet_gateway" {
 resource "aws_route" "private_nat_gateway" {
   route_table_id         = "${element(aws_route_table.private.*.id, count.index)}"
   destination_cidr_block = "0.0.0.0/0"
-  nat_gateway_id         = "${element(aws_nat_gateway.natgw.*.id, count.index)}"
-  count                  = "${length(var.private_subnets) * lookup(map(var.enable_nat_gateway, 1), "true", 0)}"
+  nat_gateway_id         = "${element(aws_nat_gateway.natgw.*.id, count.index % length(var.azs))}"
```

Comment:
Thanks for the feedback. I've added a commit in my repo which removes the modulo.

I had also added a commit which changed the number of private routes to be dependent on AZs and not the number of private subnets.. I realize these changes may just be my personal preference, but it shouldn't change anything for deployments already using a single private subnet per AZ.